### PR TITLE
Change flux_handle_destroy prototype to avoid potential memory corruption, fix small leak in flux cmd help

### PR DESCRIPTION
**Pull request by grondo**
As discussed in #1062, we were seeing memory corruption from an error path in ssh connector because of the order of destruction of connector implementation object and flux handle object. Since `flux_handle_destroy` wanted to nullify its argument, but called the implementation's destroy method first, this could result in writing to freed memory if the handle was stored *within* the implementation (as in ssh connector).

Long story short - it is probably simplest to avoid the "destroy nullifies its argument" idiom in this case, as it is rarely used in the rest of flux API, and we can thereby avoid other accidental cases of memory corruption like this in the future.

I also found a small memory leak in cmd/cmdhelp.c that I threw in here (after some valgrind runs)

### CODE REVIEW

**Comment by garlick** [comment]
Thanks for fixing that @grondo.  Looks good to me.
